### ISSUE TITLE
Config file parser bugs

### ISSUE THREAD

**Issue by pwablito** [pwablito/hpc_health_check#4]
Expanding types is not fully stable. For example, the following config crashes on the validation stage:
```
{
    "connections": [{
        "type": "connection.local.ssh.SSHConnection",
        "config": "config.connection.connection.SSHConnectionConfiguration(\"tmpuser\", \"superbadpassword\", \"localhost\")"
    }, {
        "type": "connection.local.local.LocalConnection",
        "config": "config.connection.connection.LocalConnectionConfiguration()"
    }],
    "checks": [{
        "name": "read_check",
        "check": "check.disk.read.ReadCheck",
        "config": "config.check.disk.read.ReadCheckConfiguration(16, 256, 16, 5)"
    }]
}
```
Additionally, the parser is not very good at reading arguments from function calls cleanly. The configuration mentioned above includes an object instantiation via a constructor, but doesn't pass any arguments to it (`LocalConnectionConfiguration()`). This also gets read in as a string, which is not the intended functionality.

In summary the following are high priorities:
- [ ] Fix parsing and importing of type objects
- [x] Fix parsing and importing of constructed classes by repairing and extending the argument parser to work on any number of arguments (including 0)
